drivers/gpio/gpio_mspm0.c

```diff
@@ -0,0 +1,311 @@
+}
+
+static int gpio_mspm0_port_clear_bits_raw(const struct device *port, uint32_t mask)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	DL_GPIO_clearPins(config->base, mask);
+
+	return 0;
+}
+
+static int gpio_mspm0_port_toggle_bits(const struct device *port, uint32_t mask)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	DL_GPIO_togglePins(config->base, mask);
+
+	return 0;
+}
+
+static int gpio_mspm0_pin_interrupt_configure(const struct device *port, gpio_pin_t pin,
+					      enum gpio_int_mode mode, enum gpio_int_trig trig)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	/* Config interrupt */
+	switch (mode) {
+	case GPIO_INT_MODE_DISABLED:
+		DL_GPIO_clearInterruptStatus(config->base, BIT(pin));
+		DL_GPIO_disableInterrupt(config->base, BIT(pin));
+		break;
+	case GPIO_INT_MODE_EDGE:
+		uint32_t polarity = 0x00;
+
+		if (trig & GPIO_INT_TRIG_LOW) {
+			polarity |= BIT(0);
+		}
+		if (trig & GPIO_INT_TRIG_HIGH) {
+			polarity |= BIT(1);
+		}
+		if (pin < 16) {
+			DL_GPIO_setLowerPinsPolarity(config->base, polarity << (2 * pin));
+		} else {
+			DL_GPIO_setUpperPinsPolarity(config->base, polarity << (2 * (pin - 16)));
+		}
+		DL_GPIO_clearInterruptStatus(config->base, BIT(pin));
+		DL_GPIO_enableInterrupt(config->base, BIT(pin));
+		break;
+	case GPIO_INT_MODE_LEVEL:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int gpio_mspm0_manage_callback(const struct device *port, struct gpio_callback *callback,
+				      bool set)
+{
+	struct gpio_mspm0_data *data = port->data;
+
+	return gpio_manage_callback(&data->callbacks, callback, set);
+}
+
+static uint32_t gpio_mspm0_get_pending_int(const struct device *port)
+{
+	const struct gpio_mspm0_config *config = port->config;
+
+	return DL_GPIO_getPendingInterrupt(config->base);
+}
+
+static void gpio_mspm0_isr(const struct device *port)
+{
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpioa), okay)
+	const struct device *dev_a = DEVICE_DT_GET(GPIOA_NODE);
+	struct gpio_mspm0_data *data_a = dev_a->data;
+	const struct gpio_mspm0_config *config_a = dev_a->config;
+
+	uint32_t status_a = DL_GPIO_getRawInterruptStatus(config_a->base, 0xFFFFFFFF);
+
+	DL_GPIO_clearInterruptStatus(config_a->base, status_a);
+
+	gpio_fire_callbacks(&data_a->callbacks, dev_a, status_a);
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(gpioa), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(gpiob), okay)
+	const struct device *dev_b = DEVICE_DT_GET(GPIOB_NODE);
+	struct gpio_mspm0_data *data_b = dev_b->data;
+	const struct gpio_mspm0_config *config_b = dev_b->config;
+
+	uint32_t status_b = DL_GPIO_getRawInterruptStatus(config_b->base, 0xFFFFFFFF);
+
+	DL_GPIO_clearInterruptStatus(config_b->base, status_b);
+
+	gpio_fire_callbacks(&data_b->callbacks, dev_b, status_b);
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(gpiob), okay) */
+}
+
+static bool init_irq = true;
+
+static int gpio_mspm0_init(const struct device *port)
+{
+	/* Powering up of GPIOs is part of soc.c */
+
+	if (init_irq) {
```

Comment:
interrupt cannot be enabled for both banks together?

Reply:
ok, both banks fall under same interrupt group. But for the init and irq to work, need to use `DT_INST_FOREACH_STATUS_OKAY`.